### PR TITLE
Add comment and test for adding not null column in Iceberg

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add comment and test for adding not null column in Iceberg

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Review by findepi — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -1358,7 +1358,9 @@ private static Term toIcebergTerm(Schema schema, PartitionField partitionField)
     @Override
     public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnMetadata column)
     {
-        // TODO https://github.com/trinodb/trino/issues/13587 Allow NOT NULL constraint when the table is empty
+        // Spark doesn't support adding a NOT NULL column to Iceberg tables
+        // Also, Spark throws an exception when reading the table if we add such columns and execute a rollback procedure
+        // because they keep returning the latest table definition even after the rollback https://github.com/apache/iceberg/issues/5591
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:1363]: Right, added the code comment.
